src/main/java/com/p6spy/engine/spy/P6DataSource.java

```diff
@@ -284,19 +286,35 @@ public Connection getConnection() throws SQLException {
     if (realDataSource == null) {
       bindDataSource();
     }
+    JdbcEventListener jdbcEventListener = P6Core.getJdbcEventListener();
     final long start = System.nanoTime();
-    final Connection connection = ((DataSource) realDataSource).getConnection();
-    return P6Core.wrapConnection(connection, ConnectionInformation.fromDataSource(realDataSource, connection, System.nanoTime() - start));
+    try {
+      final Connection conn = ((DataSource) realDataSource).getConnection();
+      ConnectionInformation connectionInformation = ConnectionInformation.fromDataSource(realDataSource, conn, System.nanoTime() - start);
+      jdbcEventListener.onAfterGetConnection(connectionInformation, null);
+      return ConnectionWrapper.wrap(conn, jdbcEventListener, connectionInformation);
+    } catch (SQLException e) {
+      jdbcEventListener.onAfterGetConnection(ConnectionInformation.fromDataSource(realDataSource, null, System.nanoTime() - start), e);
```

Comment:
Only one thing that could throw `SQLException` is method `getConnection`, so `onAfterGetConnection` from try block will not be called. Only if user defined `onAfterGetConnection` itself will throw that exception